js/src/common/Session.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * The `Session` class defines the current user session. It stores a reference
+ * to the current authenticated user, and provides methods to log in/out.
+ */
+import User from './models/User';
+import { RequestOptions } from './Application';
+
+export type LoginParams = {
+  /**
+   * The username/email
+   */
+  identification: string;
+
+  /**
+   * Password
+   */
+  password: string;
+};
+
+export default class Session {
+  /**
+   * The current authenticated user.
+   */
+  public user: User | null;
```

Comment:
We don't seem to explicitly state `public` in most of our other TS files, but I'm not inherently opposed to it.

I'd like to hear others' opinions on it.